zcash_client_sqlite/src/wallet/init/migrations/add_transaction_views.rs

```diff
@@ -0,0 +1,356 @@
+        )?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use rusqlite::{self, params, NO_PARAMS};
+    use tempfile::NamedTempFile;
+    use zcash_primitives::{
+        consensus::{BlockHeight, BranchId, Network},
+        legacy::Script,
+        transaction::{
+            components::{
+                transparent::{self, Authorized, OutPoint},
+                Amount, TxIn, TxOut,
+            },
+            TransactionData, TxVersion,
+        },
+    };
+
+    use crate::{
+        tests::{self},
+        wallet::init::{init_wallet_db, init_wallet_db_internal, WalletMigration2},
+        WalletDb,
+    };
+
+    #[test]
+    fn transaction_views() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_wallet_db(&mut db_data, None).unwrap();
+
+        db_data.conn.execute_batch(
+            "INSERT INTO accounts (account, ufvk) VALUES (0, '');
+            INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (0, 0, 0, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (0, 0, '');
+
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, address, value)
+            VALUES (0, 2, 0, 0, '', 2);
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, address, value, memo)
+            VALUES (0, 2, 1, 0, '', 3, X'61');
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, address, value, memo)
+            VALUES (0, 2, 2, 0, '', 0, X'f600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000');
+
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change, memo)
+            VALUES (0, 0, 0, '', 5, '', 'a', false, X'62');
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change, memo)
+            VALUES (0, 1, 0, '', 7, '', 'b', true, X'63');",
```

Comment:
This test data is still inconsistent:
- `sent_notes` contains notes with `(tx, output_pool, output_index)` of `(0, 2, 0)`, `(0, 2, 1)`, `(0, 2, 2)`.
- `received_notes ` contains notes with `(tx, output_pool, output_index)` of `(0, 2, 0)`, `(0, 2, 1)` (because the entire `received_notes` table is implicitly Sapling).
- The two notes that appear in both `sent_notes` and `received_notes` have different values and memos.

https://github.com/zcash/librustzcash/pull/620#discussion_r961133991 doesn't apply, because if it _did_, we would have no way to ever use or migrate the `received_notes` table. We _have_ to treat all notes in the (current) `received_notes` table as Sapling notes, because that's all that has ever been added to that table, and that is how we interpret the notes in that table when spending notes. When we add Orchard support, we'll need to migrate the table to have an `output_pool` column like `sent_notes` does (or do some other more involved migration to deduplicate notes if we decide that's a more useful data model, which in any case will require treating the existing notes as Sapling notes).